### ISSUE TITLE
FSharp.Compiler.Interactive.Settings distribution

### ISSUE THREAD

**Issue by auduchinok** [Microsoft/visualfsharp#5963]
FSharp.Compiler.Interactive.Settings.dll is not currently bundled in FCS package, making `fsi` object not resolved in scripts when using FCS.

Possible solutions I see are to either add it to the FCS package or to produce a separate package FCS clients should reference.

cc @KevinRansom

**Comment by auduchinok** [comment]
Another solution would be to pass the path to a chosen fsi directory to FCS.

**Comment by KevinRansom** [comment]
@auduchinok,

And I threw the potato straight back at him :-)
I am happy for us to own the FCS build and distribution, before that can happen there is some coordination with fsharp.org that needs to happen, as well as probably a reorganization of the FCS folder and projects once we agree to own it.  However, in the mean time, it really is Don's baby.

Kevin

**Comment by dsyme** [comment]
> @dsyme, packaging fcs is your thang my friend.

@KevinRansom This small DLL is not part of FCS (despite the name) but rather part of the F# scripting programming model, i.e. is implicitly referenced by every F# script.  Think of there being an implicit `#r "nuget: FSharp.Compiler.Interactive.Settings, ..."` at the top of each script file, except there is no actual nuget for this, just a DLL found by magic rules.

So my gut feeling is that going forward that we should just have an official Microsoft nuget for this, and that that model fits your view of the world better (i.e. that there should be less magic, and more nuget in `#r` resolution).

**Comment by cartermp** [comment]
I agree that we should create a package for this (Microsoft owner, signed with Microsoft keys, etc.)
